test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -130,21 +162,106 @@ public void TwoConnections()
         }
 
         [Test]
+        public void TwoConsecutiveConnections()
```

Comment:
This usage pattern is common with NHibernate, when a single transaction scope span many NHibernate session consecutively used. If in such case there is no transaction promotion to distributed, this is better.